tests/test_wagon.py

```diff
@@ -0,0 +1,686 @@
+
+    def test_create_zip_formatted_wagon_from_zip(self):
+        self.archive_name = wagon._set_archive_name(
+            TEST_PACKAGE_NAME,
+            TEST_PACKAGE_VERSION,
+            self.python_versions,
+            self.platform)
+        params = {
+            '-v': None,
+            '-f': None,
+            '-t': 'zip'
+        }
+        result = _invoke_click('create_wagon', [TEST_ZIP], params)
+        metadata = self._test(result)
+        self.assertEqual(metadata['package_source'], TEST_ZIP)
+
+    def test_create_archive_from_pypi_with_additional_wheel_args(self):
+        fd, reqs_file_path = tempfile.mkstemp()
+        os.write(fd, 'virtualenv==13.1.2')
+        params = {
+            '-v': None,
+            '-f': None,
+            '-a': '-r {0}'.format(reqs_file_path),
+            '--keep-wheels': None
+        }
+        result = _invoke_click('create_wagon', [TEST_PACKAGE], params)
+        metadata = self._test(
+            result=result,
+            expected_number_of_wheels=6)
+        self.assertEqual(metadata['package_source'], TEST_PACKAGE)
+        self.assertIn(
+            'virtualenv-13.1.2-py2.py3-none-any.whl',
+            metadata['wheels'])
+        os.close(fd)
+
+    def test_create_archive_in_destination_dir_from_pypi_latest(self):
+        temp_dir = tempfile.mkdtemp()
+        shutil.rmtree(temp_dir)
+        package = 'wheel'
+        params = {
+            '-v': None,
+            '-f': None,
+            '-o': temp_dir
+        }
+        try:
+            self.platform = 'any'
+            pypi_version = \
+                wagon._get_package_info_from_pypi(package)['version']
+            self.archive_name = \
+                wagon._set_archive_name(
+                    package,
+                    pypi_version,
+                    self.python_versions,
+                    self.platform)
+            result = _invoke_click('create_wagon', [package], params)
+            self.assertEqual(result.exit_code, 0)
+            metadata = wagon.show(os.path.join(temp_dir, self.archive_name))
+            self.platform = 'linux_x86_64'
+            self.assertEqual(pypi_version, metadata['package_version'])
+        finally:
+            shutil.rmtree(temp_dir)
+
+    def test_create_with_requirements(self):
+        test_package = os.path.join('tests', 'resources', 'test-package')
+        archive_path = wagon.create(
+            source=test_package,
+            force=True,
+            with_requirements=True,
+            verbose=True)
+        self.archive_name = os.path.basename(archive_path)
+        self.platform = 'any'
+        metadata = wagon.show(self.archive_name)
+        wheel_names = [whl.split('-')[0] for whl in metadata['wheels']]
+        self.assertIn('wheel', wheel_names)
+        self.assertIn('test_package', wheel_names)
+
+    def test_create_archive_from_url_with_requirements(self):
+        if wagon.IS_WIN:
```

Comment:
to make the test code cleaner take advantage of testtools.skipIf decorator. like so:
@testtools.skipIf(wagon.IS_WIN,'Due to...')